### ISSUE TITLE
command-cooldown-persistence doesn't work.

### ISSUE THREAD

**Issue by andrewkm** [EssentialsX/Essentials#1984]
command-cooldown-persistence: true

Simply doesn't work.
This is an example of how userdata looks:

```
  command-cooldowns:
  - pattern: repair( .*)?
    expiry: 1525095799134
```

On login, you can use a command which should be on a cooldown, and it just goes through and resets the above unix time to a new one. The cooldown works fine while the server is turned on, and obeys the expiry timestamp, however once a restart goes through, everyones cooldowns work again, and expiry isn't followed.

**Comment by mdcfe** [comment]
Could you give us the full output of `/ess version` as requested in the issue template?

**Comment by andrewkm** [comment]
nah too lazy, we use our own ess, but tested the issue on ur latest compile and it's broken as well.
I'll let you guys know when we got it fixed and submit a pull. just busy now and thought id throw this into ur issues

**Comment by mdcfe** [comment]
First I remember hearing about this but thanks for the heads up.

**Comment by JamieSinn** [comment]
Some logs I've put in for debug: 

[17:19:49] [Server thread/INFO]: xWILLISxFREAKx0 issued server command: /repair
[17:19:49] [Server thread/INFO]: [Essentials] Checking player xWILLISxFREAKx0's cooldowns on commands.
[17:19:49] [Server thread/INFO]: [Essentials] CommandCooldowns Map size: 0
[17:19:49] [Server thread/INFO]: [Essentials] checking if xWILLISxFREAKx0 has essentials.commandcooldowns.bypass - false
[17:19:49] [Server thread/INFO]: [Essentials] Command: repair /repair
[17:19:49] [Server thread/INFO]: [Essentials] Cooldown was not found.

Basically what this is showing is that the first time a player executes the command that is listed in cooldown in their userfile, it ignores it - the cooldown map is empty, therefore it has no cooldowns.

You can therefore see that the map is not being updated properly on player login after server start.
@md678685

**Comment by JamieSinn** [comment]
Then, re-doing the command after it fails: 
[17:19:50] [Server thread/INFO]: xWILLISxFREAKx0 issued server command: /repair
[17:19:50] [Server thread/INFO]: [Essentials] Checking player xWILLISxFREAKx0's cooldowns on commands.
[17:19:50] [Server thread/INFO]: [Essentials] CommandCooldowns Map size: 1
[17:19:50] [Server thread/INFO]: [Essentials] checking if xWILLISxFREAKx0 has essentials.commandcooldowns.bypass - false

**Comment by mdcfe** [comment]
@JamieSinn Thanks for looking into this. I'm assuming this means the map isn't loaded properly when it should have been - I don't remember whether this is the job of the PlayerLoginEvent listener or UserData.

**Comment by JamieSinn** [comment]
Currently it's done by UserData, but it should be done in PlayerLoginListener

**Comment by andrewkm** [comment]
@md678685 any ideas on what to do with this one. Would be nice to get this specific cool down working.

**Comment by andrewkm** [comment]
@md678685 any updates here on this one?

**Comment by andrewkm** [comment]
Any updates on this @md678685 - still broken in essentialsx.
Cooldowns work only until a server restart, at which point, even though they all remain in the userdata, they act as if they have been reset.

**Comment by mdcfe** [comment]
@andrewkm Please don't bump threads. I will update threads when I have something to add.

**Comment by andrewkm** [comment]
Ok bb

**Comment by mdcfe** [comment]
@andrewkm Is this still an issue?

**Comment by andrewkm** [comment]
I’ve fixed it long ago.
Was going over my old tickets on github.
It is however broken on EssentialsX though.
